include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -650,6 +651,26 @@ def ToBuiltinBoolOp : MooreOp<"to_builtin_bool", [Pure]> {
   }];
 }
 
+def TimeToIntOp : MooreOp<"time_to_int", [Pure]> {
+  let summary = "Convert a time type to an integer number of femtoseconds";
+  let arguments = (ins TimeType:$input);
+  let results = (outs FourValuedI64:$result);
+  let assemblyFormat = "$input attr-dict";
+  let hasFolder = 1;
+}
+
+def IntToTimeOp : MooreOp<"int_to_time", [Pure]> {
```

Comment:
Can either of these either fail or are all inputs valid?

Reply:
All inputs are valid. Sadly, time values are just unit-less integers in Verilog, so any conversion issues or unit mismatches are basically mandated by the standard 🤦‍♂️